src/ui/components/VPNExpandableAppList.qml

```diff
@@ -10,140 +10,118 @@ import QtGraphicalEffects 1.15
 import Mozilla.VPN 1.0
 import "../themes/themes.js" as Theme
 
-VPNClickableRow {
-    id: appRow
+ColumnLayout{
+    id: appListContainer
 
-    property var listVisible : true
     property var listModel: undefined
     property var header: ""
     property var description: ""
     property var actionText: ""
     property var onAction: ()=>{}
     property var isEnabled: true
-
-
-    state: (listVisible && isEnabled && applist.count > 0) ? "list-visible" : "list-hidden"
+    property var isListVisible : true && applist.count > 0
     opacity: isEnabled ? 1 : 0.5
-
-    Keys.onReleased: if (event.key === Qt.Key_Space) handleKeyClick()
-    handleMouseClick: function() { listVisible = !listVisible; }
-    handleKeyClick: function() { listVisible = !listVisible; }
-    clip: true
-    canGrowVertical: true
-    accessibleName: name
-    states: [
-        State {
-            name: "list-hidden"
-            PropertyChanges {
-                target: toggleArrow
-                rotation: -90
-
-            }
-            PropertyChanges {
-                target: appRow
-                height: appRowHeader.height
-            }
-
-        },
-        State {
-            name: "list-visible"
-
-            PropertyChanges {
-                target: toggleArrow
-                rotation: 0
-
-            }
-            PropertyChanges {
-                target: appRow
-                height: appRowHeader.height +applist.contentHeight + applist.anchors.topMargin
-            }
-
-        }
-    ]
-
-
-    RowLayout {
-        id: appRowHeader
-        anchors.top: parent.top
-        anchors.left: parent.left
-        width: appRow.width
-        spacing: 0
-
-        VPNIcon {
-            id: toggleArrow
-            Layout.leftMargin: 15 - appRow.anchors.leftMargin
-            Layout.rightMargin: 15
-            source: "../resources/arrow-toggle.svg"
-            transformOrigin: Image.Center
-            smooth: true
-            opacity: applist.count > 0 ?1:0;
-        }
-        ColumnLayout {
-            Layout.alignment: Qt.AlignLeft
-            Layout.fillWidth: true
+    property var isActionEnabled: isEnabled && applist.count > 0
+    height: appRowHeader.height + (isListVisible  ? applist.height + defaultMargin : 0)
+
+    VPNClickableRow {
+        id: appRow
+        Keys.onReleased: if (event.key === Qt.Key_Space) handleKeyClick()
+        handleMouseClick: function() { isListVisible = !isListVisible && applist.count > 0; }
+        handleKeyClick: function() { isListVisible = !isListVisible && applist.count > 0; }
+        clip: true
+        canGrowVertical: true
+        accessibleName: name
+        width: parent.width
+        height: appRowHeader.height
+        Layout.preferredHeight: appRowHeader.height
```

Comment:
Same thing here with the binding loop. This can be `Layout.preferredHeight: Theme.rowHeight * 1.5`